controllers/discovery/initDiscoveryUserData.js

```diff
@@ -59,10 +61,10 @@ const InitDiscoveryUserData = async (req, res) => {
         ON A.user_id = B.user_id_followed AND B.user_id_follower = :userId
         WHERE A.user_id != :userId AND A.is_anonymous = false AND A.is_banned = false
         ORDER BY
+        COALESCE(B.user_id_follower, '') DESC,
         COALESCE(A.karma_score, 0) DESC,
         COALESCE(CommonUsers.common, -1) DESC, 
-        COALESCE(CommonUsers.user_match, -1) DESC,
-        COALESCE(B.user_id_follower, '') DESC
+        COALESCE(CommonUsers.user_match, -1) DESC
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The change in the order of the `ORDER BY` clause could potentially affect the performance of the query. The new ordering prioritizes sorting by `B.user_id_follower` which is a string field, and string comparisons are generally slower than numeric comparisons. If the `user_id_follower` field has high cardinality (i.e., many unique values), this could slow down the query significantly.

However, without knowing more about the data distribution and the database schema, it's hard to say definitively whether this will be an issue. It would be advisable to test this query with production-like data to ensure that it performs acceptably.

Also, consider adding indexes on the fields used in the `ORDER BY` clause if they don't exist already, as this can improve the performance of the sort operation.


<!-- This is an auto-generated comment by OSS CodeRabbit -->